tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_persistence.c

```diff
@@ -21,18 +22,242 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+	.addr = test_va[1].addr,
+	.uuid = test_va[1].uuid,
+	.app_idx = test_appkeys[1].idx,
+	.cred_flag = true,
+	.ttl = 5,
+	.period = BT_MESH_PUB_PERIOD_10SEC(1),
+	.transmit = BT_MESH_TRANSMIT(2, 20),
+};
+
+static const struct bt_mesh_cfg_mod_pub disabled_mod_pub_params = {
+	.addr = 0,
+	.uuid = NULL,
+	.app_idx = 0,
+	.cred_flag = false,
+	.ttl = 0,
+	.period = 0,
+	.transmit = 0,
+};
+
+static ssize_t test_preset = -1;
+static struct {
+	struct bt_mesh_cfg_mod_pub pub_params;
+
+	size_t appkeys_count;
+	uint16_t appkeys[CONFIG_BT_MESH_MODEL_KEY_COUNT];
+
+	size_t subs_count;
+	uint16_t subs[CONFIG_BT_MESH_MODEL_GROUP_COUNT];
+
+	size_t mod_data_len;
+} test_access_presets[][2] = {
+	/* Default */
+	{
+		/* SIG model. */
+		{
+			.pub_params = test_mod_pub_params,
+			.appkeys_count = 2, .appkeys = { test_appkeys[0].idx, test_appkeys[1].idx },
+			.subs_count = 2, .subs = { test_group[0], test_va[0].addr },
+			.mod_data_len = sizeof(test_mod_data),
+		},
+
+		/* Vendor model. */
+		{
+			.pub_params = test_vnd_mod_pub_params,
+			.appkeys_count = 2, .appkeys = { test_appkeys[0].idx, test_appkeys[1].idx },
+			.subs_count = 2, .subs = { test_group[1], test_va[1].addr },
+			.mod_data_len = sizeof(vnd_test_mod_data),
+		},
+	},
+
+	/* After subscription overwrite */
+	{
+		/* SIG model. */
+		{
+			.pub_params = test_mod_pub_params,
+			.appkeys_count = 2, .appkeys = { test_appkeys[0].idx, test_appkeys[1].idx },
+			.subs_count = 1, .subs = { test_group[1] },
+			.mod_data_len = sizeof(test_mod_data),
+		},
+
+		/* Vendor model. */
+		{
+			.pub_params = test_vnd_mod_pub_params,
+			.appkeys_count = 2, .appkeys = { test_appkeys[0].idx, test_appkeys[1].idx },
+			.subs_count = 1, .subs = { test_va[0].addr },
+			.mod_data_len = sizeof(vnd_test_mod_data),
+		},
+	},
+
+	/* After remove */
+	{
+		/* SIG model. */
+		{
+			.pub_params = disabled_mod_pub_params,
+			.appkeys_count = 0, .appkeys = {},
+			.subs_count = 0, .subs = {},
+			.mod_data_len = 0,
+		},
+
+		/* Vendor model. */
+		{
+			.pub_params = disabled_mod_pub_params,
+			.appkeys_count = 0, .appkeys = {},
+			.subs_count = 0, .subs = {},
+			.mod_data_len = 0,
+		},
+	},
+};
+
+static const struct {
+	uint8_t beacon;
+	uint8_t ttl;
+	uint8_t gatt_proxy;
+	uint8_t friend;
+	uint8_t net_transmit;
+	struct {
+		enum bt_mesh_feat_state state;
+		uint8_t transmit;
+	} relay;
+} test_cfg_presets[] = {
+	{
+		.beacon = 1,
+		.ttl = 12,
+		.gatt_proxy = 1,
+		.friend = 1,
+		.net_transmit = BT_MESH_TRANSMIT(3, 20),
+		.relay = { .state = BT_MESH_FEATURE_ENABLED, .transmit = BT_MESH_TRANSMIT(2, 20) },
+	},
+	{
+		.beacon = 0,
+		.ttl = 0,
+		.gatt_proxy = 0,
+		.friend = 0,
+		.net_transmit = BT_MESH_TRANSMIT(1, 30),
+		.relay = { .state = BT_MESH_FEATURE_ENABLED, .transmit = BT_MESH_TRANSMIT(1, 10) },
+	},
+};
+
+static void test_args_parse(int argc, char *argv[])
+{
+	for (size_t i = 0; i < argc; i++) {
+		if (strstr(argv[i], "test-preset=")) {
```

Comment:
You could use the argument parsing utilities provided in the bsim Util library.
For this particular check, this is a bit more robust:
https://github.com/BabbleSim/base/blob/master/libUtilv1/src/bs_cmd_line.c#L54

Or if you wanted to go the whole way, you could use:
https://github.com/BabbleSim/base/blob/master/libUtilv1/src/bs_cmd_line.c#L511

Reply:
Replaced with `bs_args_parse_all_cmd_line`.